tuning/sft_trainer.py

```diff
@@ -292,23 +296,63 @@ def main(**kwargs):  # pylint: disable=unused-argument
         choices=["pt", "lora", None, "none"],
         default="none",
     )
+    parser.add_argument(
+        "--exp_metadata",
+        type=str,
+        default=None,
+        help='Pass a json string representing K:V pairs to be associated\
+              to the tuning run in the tracker. e.g. \'{"gpu":"A100-80G"}\'',
+    )
     (
         model_args,
         data_args,
         training_args,
         trainer_controller_args,
         lora_config,
         prompt_tuning_config,
-        peft_method,
+        file_logger_config,
+        aim_config,
+        additional,
         _,
     ) = parser.parse_args_into_dataclasses(return_remaining_strings=True)
-    if peft_method.peft_method == "lora":
+
+    logger = logging.get_logger("__main__")
+
+    peft_method = additional.peft_method
+    if peft_method == "lora":
         tune_config = lora_config
-    elif peft_method.peft_method == "pt":
+    elif peft_method == "pt":
         tune_config = prompt_tuning_config
     else:
         tune_config = None
-    train(model_args, data_args, training_args, tune_config, trainer_controller_args)
+
+    # extra metadata passed via client
+    metadata = None
+    if additional.exp_metadata is not None:
+        try:
+            metadata = json.loads(additional.exp_metadata)
+            if metadata is None or not isinstance(metadata, Dict):
+                logger.warning(
+                    "metadata cannot be converted to simple k:v dict ignoring"
+                )
+                metadata = None
+        except ValueError as e:
+            logger.error(
+                "failed while parsing extra metadata. pass a valid json %s", repr(e)
+            )
+
+    combined_tracker_configs = {"file_logger": file_logger_config, "aim": aim_config}
+
+    train(
+        model_args=model_args,
+        data_args=data_args,
+        train_args=training_args,
```

Comment:
I still have doubt on what default tracker is passed with training_args here. wby default if users use command line, we want the logs written to log file like we were earlier.

Reply:
oh I see you added default tracker inside code and not through arguments